types/src/error.rs

```diff
@@ -73,7 +73,7 @@ impl<'a> ErrorObject<'a> {
 	}
 
 	/// Create a new [`ErrorObject`] with optional data.
-	pub fn borrowed(code: i32, message: &'a impl AsRef<str>, data: Option<&'a RawValue>) -> ErrorObject<'a> {
+	pub fn borrowed(code: i32, message: &'a (impl AsRef<str> + ?Sized), data: Option<&'a RawValue>) -> ErrorObject<'a> {
```

Comment:
Good catch; `&'a AsRef<str>` means that `str == AsRef<str>` if you pass in `&str`, so indeed we need the `?Sized` here. (oftentimes we can just accept `AsRef<str>` ie no reference, if we don't care about the lifetime).

I do think the whole thing should probably just be simpler and accept `&'a str` though; I suspect doing so would make it no harder for users and make the type simpler to follow here!

Reply:
yeah, please make this `message: &'a str` or just `message: impl AsRef<str>` but unfortunately a breaking change

Reply:
We can't do the latter (` message: impl AsRef<str>`) because we need to capture/use the lifetime from the `str` (I tried it first :D) but yeah just `message: &'a str` makes sense to me!